packages/@lwc/ssr-compiler/src/compile-template/element.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { builders as b, is } from 'estree-toolkit';
+import { HTML_NAMESPACE, isVoidElement } from '@lwc/shared';
+import { esTemplateWithYield } from '../estemplate';
+import { irChildrenToEs } from './ir-to-es';
+import { bImportHtmlEscape, importHtmlEscapeKey, cleanStyleAttrVal } from './shared';
+
+import type {
+    Attribute as IrAttribute,
+    Expression as IrExpression,
+    Element as IrElement,
+    Literal as IrLiteral,
+    Property as IrProperty,
+} from '@lwc/template-compiler';
+import type {
+    BlockStatement as EsBlockStatement,
+    Expression as EsExpression,
+    Statement as EsStatement,
+} from 'estree';
+import type { Transformer } from './types';
+
+const bYield = (expr: EsExpression) => b.expressionStatement(b.yieldExpression(expr));
+const bConditionalLiveYield = esTemplateWithYield<EsBlockStatement>`
+    {
+        const attrOrPropValue = ${is.expression};
+        const valueType = typeof attrOrPropValue;
+        if (attrOrPropValue && (valueType === 'string' || valueType === 'boolean')) {
+            yield ' ' + ${is.literal};
+            if (valueType === 'string') {
+                yield '="' + htmlEscape(attrOrPropValue, true) + '"';
+            }
+        }
+    }
+`;
+
+function yieldAttrOrPropLiteralValue(name: string, valueNode: IrLiteral): EsStatement[] {
+    const { value, type } = valueNode;
+    if (typeof value === 'string') {
+        const yieldedValue = name === 'style' ? cleanStyleAttrVal(value) : value;
+        return [bYield(b.literal(` ${name}="${yieldedValue}"`))];
+    } else if (typeof value === 'boolean') {
+        return [bYield(b.literal(` ${name}`))];
+    }
+    throw new Error(`Unknown attr/prop literal: ${type}`);
```

Comment:
Will never reach this. `value` is of type `stirng | boolean`

Reply:
True. I think I like having the guard for cases when we're getting lazy about typing and we're casting or using `any` somewhere.

Reply:
Too late now, but it's a bit weird that the error is an unknown `value`, but the message is refers to `type`. `type` is always `"Literal"`, so the error message wouldn't be super helpful here if we accidentally pass, for example, a numeric literal or something.